src/util/psbt/mod.rs

```diff
@@ -181,8 +185,223 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Signs the input at `input_index` with private key `sk`.
+    pub fn sign<C: Signing>(&mut self, sk: &PrivateKey, input_index: usize, secp: &Secp256k1<C>) -> Result<(), SighashError> {
+        self.check_index_is_within_bounds(input_index)?;
+
+        let mut cache = SighashCache::new(&self.unsigned_tx);
+        let (msg, sighash_ty) = self.sighash(input_index, &mut cache, None)?;
+
+        let sig = secp.sign_ecdsa(&msg, &sk.inner);
+
+        let mut final_signature = Vec::with_capacity(75);
+        final_signature.extend_from_slice(&sig.serialize_der());
+        // FIXME: Is this ok? Can a non-standard sighash type bigger than a u8 sneek in here?
+        final_signature.push(sighash_ty.to_u32() as u8);
```

Comment:
An explicit cast will truncate the value as you expect

```rs
    // 1000 - 256 - 256 - 256 = 232
    // Under the hood, the first 8 least significant bits (LSB) are kept,
    // while the rest towards the most significant bit (MSB) get truncated.
    println!("1000 as a u8 is : {}", 1000 as u8);
    // -1 + 256 = 255
    println!("  -1 as a u8 is : {}", (-1i8) as u8);

    // For positive numbers, this is the same as the modulus
    println!("1000 mod 256 is : {}", 1000 % 256);
```
https://doc.rust-lang.org/rust-by-example/types/cast.html

Reply:
Thanks! Today I learned something. I'll remove the `FIXME`, the cast will introduce a clippy warning I believe but since we are not linting properly yet I'll fix that another day.